src/io/serialization/v2.0.0/serialize.ts

```diff
@@ -64,43 +53,24 @@ function convert(shareableDangerous: Record<string, unknown>): Shareable {
   }
 }
 
-function validateMore(shareable: Shareable) {
-  const { scenarioData, ageDistributionData, severityDistributionData } = shareable
-
-  if (scenarioData.data.population.ageDistributionName !== ageDistributionData.name) {
-    throw new DeserializationErrorValidationFailed([
-      '/scenarioData/data/population/ageDistributionName should be equal to /ageDistributionData/name',
-    ])
-  }
-
-  const ageDistributionCategories = ageDistributionData.data.map(({ ageGroup }) => ageGroup)
-  const severityCategories = severityDistributionData.data.map(({ ageGroup }) => ageGroup)
-  if (!isEqual(ageDistributionCategories, severityCategories)) {
-    throw new DeserializationErrorValidationFailed([
-      'arrays /ageDistributionData/data[] and /severityDistributionData/data[] should contain the same number of the same values for ageGroup',
-    ])
-  }
-}
-
 function deserialize(input: string): ScenarioParameters {
   const shareableDangerous = JSON.parse(input) as Record<string, unknown>
-
   validateSchema(shareableDangerous)
-
   const shareable = convert(shareableDangerous)
 
-  validateMore(shareable)
-
-  const { scenarioData, ageDistributionData, severityDistributionData } = shareable
-
-  return {
-    scenarioData: {
-      ...scenarioData,
-      data: toInternal(scenarioData.data),
+  // Migrate object to schema v2.1.0:
+  //  - Add palliative severities and set them to 0
+  const shareableNew: Shareable = {
+    ...shareable,
+    schemaVer: schemaVerNext,
+    severityDistributionData: {
+      ...shareable.severityDistributionData,
+      data: shareable.severityDistributionData.data.map((severity) => ({ ...severity, palliative: 0 })),
     },
-    ageDistributionData,
-    severityDistributionData,
   }
+
+  // Delegate to the next version of deserializer
+  return v2_1_0[schemaVerNext].deserialize(JSON.stringify(shareableNew))
 }
```

Comment:
Deserialization function for v2.0.0 deserializes the received object of v2.0.0 as before, but then also adjusts the object such that it looks like an object of v2.1.0, then passes it to the deserializer of v2.1.0.

It's a bit redundant, as most of the checks will run twice. But it's easy to implement and maintain.